From 7f69d93f777cdd18f5050aad4d18413e23e23e62 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:51:45 -0400
Subject: [fs] nfsd: remove unneeded zeroing of fields in nfsd4_proc_compound

Message-id: <1403560314-32525-164-git-send-email-bfields@redhat.com>
Patchwork-id: 84545
O-Subject: [Patch RHEL7 163/172] nfsd: remove unneeded zeroing of fields in nfsd4_proc_compound
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

The memset of resp in svc_process_common should ensure that these are
already zeroed by the time they get here.

Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 7025005d5e7715efa24c115c45f4a6b4f1a545de
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4proc.c b/fs/nfsd/nfs4proc.c
index 1bed2a9..3aab35e 100644
--- a/fs/nfsd/nfs4proc.c
+++ b/fs/nfsd/nfs4proc.c
@@ -1289,11 +1289,8 @@ nfsd4_proc_compound(struct svc_rqst *rqstp,
  xdr_reserve_space(&resp->xdr, 8 + args->taglen);
  resp->taglen = args->taglen;
  resp->tag = args->tag;
- resp->opcnt = 0;
  resp->rqstp = rqstp;
  cstate->minorversion = args->minorversion;
- cstate->replay_owner = NULL;
- cstate->session = NULL;
  fh_init(current_fh, NFS4_FHSIZE);
  fh_init(save_fh, NFS4_FHSIZE);
  /*
-- 
1.7.1